### PR TITLE
bugfix: [KOA-4876] - Datepicker modal fix

### PR DESCRIPTION
**Pull request by olliecurtis**
This PR allows the `fixedWidth` prop to become available to the datepicker so that when in the datepicker opens in a popover (when device is mobile) the calendar will fit to the correct viewport. By default the datepicker has a fixed width but always on mobile it will not.

It also changes the calendar on mobile to not use the calendar transitions as this requires a fixed width in order to calculate the transition and alignment which cannot be always determined by the device being viewed on.

| Before | After |
| -- | -- |
| Screenshot 2022-01-12 at 13 13 06 | Screenshot 2022-01-12 at 13 13 20 |

Remember to include the following changes:

- [x] `UNRELEASED.md` (See [How to write a good changelog entry](https://github.com/Skyscanner/backpack/blob/main/CHANGELOG_FORMAT.md))
- [x] `README.md`
- [x] Tests

### CODE REVIEW

File: packages/bpk-component-datepicker/src/BpkDatepicker.js [hunk 1]
```diff
@@ -331,11 +346,12 @@ BpkDatepicker.propTypes = {
 };
 
 BpkDatepicker.defaultProps = {
-  calendarComponent: BpkCalendar,
+  calendarComponent: DefaultCalender,
```
Comment (anambl) [inline packages/bpk-component-datepicker/src/BpkDatepicker.js:349]: small typo here 😄

File: packages/bpk-component-datepicker/src/BpkDatepicker.js [hunk 2]
```diff
@@ -331,11 +346,12 @@ BpkDatepicker.propTypes = {
...
   inputComponent: null,
   date: null,
   dateModifiers: BpkCalendar.defaultProps.dateModifiers,
```
[the remark this answers is not in the document]
Comment (olliecurtis) [reply packages/bpk-component-datepicker/src/BpkDatepicker.js]: Good catch!

File: packages/bpk-component-datepicker/src/BpkDatepicker.js (hunk 1 above)
Comment (olliecurtis) [reply packages/bpk-component-datepicker/src/BpkDatepicker.js]: Good catch!

**Review by anambl — APPROVED** [review]
LGTM! 🎉

**Pull request merged**